### PR TITLE
Qsv: fix timebase

### PR DESCRIPTION
**Pull request by xhaihao**

### CODE REVIEW

**Comment by xuguangxin** [comment]
could you attach some  MediaSDK document link to the first commit log?

thanks

**Review by xuguangxin — APPROVED** [review]

**Comment by xhaihao** [comment]
> could you attach some MediaSDK document link to the first commit log?

Added, thanks!

**Comment by xhaihao** [comment]
Updated the patch to check whether the time base is valid.

File: libavcodec/qsvdec.c [hunk 1]
```diff
@@ -748,6 +758,9 @@ static av_cold int qsv_decode_init(AVCodecContext *avctx)
         goto fail;
     }
 
+    if (!avctx->pkt_timebase.num)
```
Comment (xuguangxin) [inline libavcodec/qsvdec.c:761]: What's happened if just ignore msdk document just set avcodec pts to TimeStamp ?

Comment (xhaihao) [reply libavcodec/qsvdec.c:761]: You might get wrong timestamp from the SDK and the muxer will print some error messge. e.g.

$> ffmpeg -hwaccel qsv -c:v hevc_qsv  -i output.h265  -f null -
[null @ 0x561c24f6f2f0] Application provided invalid, non monotonically increasing dts to muxer in stream 0: 2 >= 2

**Comment by xhaihao** [comment]
Send to FFmpeg mailing list:
http://ffmpeg.org/pipermail/ffmpeg-devel/2021-June/281249.html
http://ffmpeg.org/pipermail/ffmpeg-devel/2021-June/281250.html

**Pull request closed without merging**